RecoHI/HiEvtPlaneAlgos/src/HiEvtPlaneFlatProducer.cc

```diff
@@ -150,52 +224,78 @@ HiEvtPlaneFlatProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetu
   using namespace std;
   using namespace reco;
 
+  bool newrun = false;
+  if(runno_ != iEvent.id().run()) newrun = true;
+  runno_ = iEvent.id().run();
+  
   //
-  //Get Centrality
+  //Get Flattening Parameters
   //
+  if(FirstEvent || newrun) {
```

Comment:
use ESWatcher  instead for assuming that the record changes only between the runs
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_each_time_a_Record